app/controllers/slack_controller.rb

```diff
@@ -12,6 +12,8 @@ def event
     end
 
     case slack_params.fetch(:event, {})[:type]
+    when "admins"
```

Comment:
So... at first I really liked that this was combined into one action, but then it occurred to me that we may run into naming conflicts between slash commands and actual slack events. For instance with this setup we can't have a slash command named `message`.  Perhaps contrived, but I think it could lead to problems.

What about routing slash commands into this controller, but to a `def command` action with similar functionality just to ensure we avoid any future name collision?

Reply:
Nooooooooooo that's how I had it before and I got feedback to treat it like other events! 😂Let me think about it; we could do a separate class or object for commands with their own switch case.

Reply:
Uh oh.. that was probably me wasn't it :(

What if... hrm... we change the route so that /slack/event and /slack/command both hit the same action (event_or_command) and then we could differentiate it in there using say `params[:type]`.  

The name conflict is really my only concern... and maybe I'm over thinking it.

Reply:
I also feel like the likelihood that we're going to want a `/message` is....limited.  Like, what would it do, that isn't like, opening a DM (and why would we do that with a slash command)?  All the other events we could subscribe to seem like: a) something we don't actually need to replicate with a slack command (`bot_changed`?) or b) something we could rename to something more specific #becauseSynonyms.  Just my thought.

Reply:
Yeah, good point. Forget I said anything :)

Reply:
![https://media.giphy.com/media/l46CnlK71u7CazMUE/giphy.gif](https://media.giphy.com/media/l46CnlK71u7CazMUE/giphy.gif)